src/GraphQLEditor/GqlEditor.jsx

```diff
@@ -1,45 +1,95 @@
 import GraphiQL from 'graphiql';
 import { buildClientSchema } from 'graphql/utilities';
+import Dropdown from '@gen3/ui-component/dist/components/Dropdown';
 import React from 'react';
 import PropTypes from 'prop-types';
-import { fetchGraphQL } from '../actions';
+import { fetchGraphQL, fetchArrangerGraphQL } from '../actions';
 import Spinner from '../components/Spinner';
 import './GqlEditor.less';
 
 const parameters = {};
 
-const GqlEditor = ({ schema }) => {
-  if (!schema) {
-    return <Spinner />; // loading
+class GqlEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedEndpointIndex: props.endpointIndex || 0,
```

Comment:
What if not `0` or `1`? will it break if `endpoint=2`?

Reply:
Probably 🙈 will add a check for that